builders/interoperability/xcm/xc-registration/self-serve-asset-registration.md

```diff
@@ -0,0 +1,194 @@
+  --asset '{"parents":1,"interior":{"X3":[{"Parachain":4},{"PalletInstance":12},{"GeneralIndex":15}]}}' \
+  --network moonbeam
+```
+
+The script will return the `assetID` you are now ready to pass to `evmForeignAssets.createForeignAsset`.
+
+### Derive the XC-20 Address
+
+Convert `assetID` to hex, left-pad it to 32 hex chars, and prepend eight `F`s as follows:
+
+```
+xc20Address = 0xFFFFFFFF + hex(assetId).padStart(32, '0')
+```
+
+The XC-20 address of xcDOT as an example can be calculated like so: 
+
+=== "Formula"
+
+    ```ts
+    const xc20Address = `0xFFFFFFFF${hex(assetId).padStart(32, "0")}`;
+    ```
+=== "Example"
+
+    ```bash
+    0xFFFFFFFF1FCACBD218EDC0EBA20FC2308C778080
+    ```
+
+
+## Generate the Encoded Call Data {: #generate-the-encoded-call-data }
+
+The snippet below shows how to build the call that needs to be sent to Moonbeam that creates the foreign asset. Save the resulting hex string because you will embed it inside a subsequent XCM `Transact` call dispatched from your sibling parachain.
+
+```typescript
+import "@moonbeam-network/api-augment";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { blake2AsHex } from "@polkadot/util-crypto";
+
+const moonbeam = await ApiPromise.create({ provider: new WsProvider(MOONBEAM_WSS) });
+
+const tx = moonbeam.tx.evmForeignAssets.createForeignAsset(
+  ASSET_ID,
+  assetLocation,
+  DECIMALS,
+  SYMBOL,
+  NAME
+);
+
+// SCALE-encoded call data (includes call index 0x3800)
+const encodedCall = tx.method.toHex();
+console.log("Encoded call data:", encodedCall);
+
+// Optional: 32-byte call hash (blake2_256)
+console.log("Call hash:", blake2AsHex(encodedCall));
+```
+
+### Dispatch the call with XCM Transact {: #dispatch-the-call-with-xcm-transact }
+
+To register your asset, wrap the SCALE‑encoded `createForeignAsset` bytes in a single `Transact` instruction executed from your parachain’s sovereign account. The basic structure of the call is outlined below:
+
+```text
+Transact {
+  originKind: SovereignAccount,
+  requireWeightAtMost: <weight>,
+  call: <encodedCall>
+}
+```
+
+Send the transact instruction via `xcmPallet.send`, targeting Parachain `2004` for Moonbeam (or `2023` for Moonriver. 
+
+```rust
+xcmPallet.send(
+  dest: { Parachain: 2004 },
+  message: VersionedXcm::V4(INSERT_TRANSACT_INSTRUCTION)
+);
+```
+
+Finally, look for the following event emitted successfully on Moonbeam:
+
+```
+EvmForeignAssets.ForeignAssetCreated(assetId, location, creator)
+```
+
+Its presence confirms the XC-20 asset is live.
+
+## Managing an Existing Foreign Asset {: #managing-an-existing-foreign-asset }
+
+After a foreign asset has been created, the following extrinsics can be used to update the foreign asset. Note that in the case of the sovereign account sending a call, the sovereign account and location must still be inside the origin. Otherwise, the only other authorized origin is `Root` from a Moonbeam governance action. 
```

Comment:
```suggestion
After a foreign asset has been created, the following extrinsics can be used to update it. Note that in the case of the sovereign account sending a call, the sovereign account and location must still be inside the origin. Otherwise, the only other authorized origin is `Root` from a Moonbeam governance action. 
```